hub-saml/src/main/java/uk/gov/ida/saml/hub/domain/EidasAttributesLogger.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.ida.saml.hub.domain;
+
+import org.joda.time.DateTime;
+import org.opensaml.core.xml.XMLObject;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.opensaml.saml.saml2.core.Attribute;
+import org.opensaml.saml.saml2.core.AttributeStatement;
+import org.opensaml.saml.saml2.core.Response;
+import org.opensaml.saml.saml2.core.Subject;
+import uk.gov.ida.saml.core.IdaConstants;
+import uk.gov.ida.saml.core.extensions.BaseMdsSamlObject;
+import uk.gov.ida.saml.core.transformers.EidasResponseAttributesHashLogger;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class EidasAttributesLogger {
+
+    private final Supplier<EidasResponseAttributesHashLogger> loggerSupplier;
+
+    public EidasAttributesLogger(Supplier<EidasResponseAttributesHashLogger> loggerSupplier) {
+        this.loggerSupplier = loggerSupplier;
+    }
+
+    public void logEidasAttributesAsHash(Assertion assertion, Response response) {
+
+        Subject subject = assertion.getSubject();
+        String pid = subject.getNameID().getValue();
+        EidasResponseAttributesHashLogger hashLogger = loggerSupplier.get();
+        hashLogger.setPid(pid);
+
+        List<AttributeStatement> attributeStatements = assertion.getAttributeStatements();
+        List<Attribute> attributes = attributeStatements.get(0).getAttributes();
```

Comment:
Worth noting that `attributeStatements` will list of exactly one `AttributeStatement`, as validated at
`uk/gov/ida/saml/core/validators/assertion/MatchingDatasetAssertionValidator.java:69`